lib/common/util.js

```diff
@@ -545,3 +545,43 @@ function normalizeArguments(globalContext, localConfig, options) {
 }
 
 util.normalizeArguments = normalizeArguments;
+
+/**
+ * Throttle requests according to a `maxApiCalls` limit.
+ *
+ * @param {function} makeRequestFn - The function that will be called.
+ * @param {object=} options - Configuration object.
+ * @param {number} options.maxApiCalls - The maximum number of API calls to
+ *     make.
+ */
+function iterator(makeRequestFn, options) {
```

Comment:
Not sure how I feel about calling this an iterator, my initial reaction was, *"like a generator?"*. How about something like `createLimiter`?

```js
var limiter = uitl.createLimiter(makeRequest, { ... });
```